service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/cmo/ReviewCMOController.java

```diff
@@ -0,0 +1,156 @@
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+
+        if (!cmosReadyForApproval.isEmpty()) {
+            Element<CaseManagementOrder> cmo = reviewCMOService.getSelectedCMO(caseData);
+
+            if (!JUDGE_REQUESTED_CHANGES.equals(caseData.getReviewCMODecision().getDecision())) {
+                caseData.getDraftUploadedCMOs().remove(cmo);
+
+                Element<CaseManagementOrder> cmoToSeal = reviewCMOService.getCMOToSeal(caseData, cmo);
+
+                DocumentReference convertedDocument = documentConversionService.convertDocument(
+                    cmoToSeal.getValue().getOrder());
+                DocumentReference sealedDocument = documentSealingService.sealDocument(convertedDocument);
+                cmoToSeal.getValue().setOrder(sealedDocument);
+
+                List<Element<CaseManagementOrder>> sealedCMOs = caseData.getSealedCMOs();
+                sealedCMOs.add(cmoToSeal);
+
+                data.put("sealedCMOs", sealedCMOs);
+            } else {
+                cmo.getValue().setStatus(RETURNED);
+                cmo.getValue().setRequestedChanges(caseData.getReviewCMODecision().getChangesRequestedByJudge());
+            }
+
+            data.put("draftUploadedCMOs", caseData.getDraftUploadedCMOs());
+            data.remove("numDraftCMOs");
+            data.remove("cmoToReviewList");
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/submitted")
+    public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+        CaseData caseDataBefore = mapper.convertValue(callbackRequest.getCaseDetailsBefore().getData(), CaseData.class);
+        CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseDataBefore.getDraftUploadedCMOs());
+
+        if (!cmosReadyForApproval.isEmpty()) {
+            if (!JUDGE_REQUESTED_CHANGES.equals(caseData.getReviewCMODecision().getDecision())) {
+                CaseManagementOrder sealed = reviewCMOService.getLatestSealedCMO(caseData.getSealedCMOs());
+
+                eventPublisher.publishEvent(new CaseManagementOrderIssuedEvent(callbackRequest, sealed));
+            } else {
+                List<Element<CaseManagementOrder>> draftCMOsBefore = caseDataBefore.getDraftUploadedCMOs();
+                List<Element<CaseManagementOrder>> draftCMOs = caseData.getDraftUploadedCMOs();
+
+                //Get the CMO that was modified (status changed from READY -> RETURNED)
+                draftCMOs.removeAll(draftCMOsBefore);
+                CaseManagementOrder cmoToReturn = draftCMOs.get(0).getValue();
```

Comment:
not sure about this, but can we use reviewCMOService.getSelectedCMO() instead of all this code?
maybe it is already cleaned up?

Reply:
I think we already cleaned up in about-to-submit, we need numDraftCMOs
but maybe we can clean in about-to-start instead? I will have a look

Reply:
dynamic list causes problems so I will leave it like this for now. I think we would have to recreate the dynamic list in submitted (getSelectedCMO() will use dynamic list if we have numDraftCMOs = MULTI)